### PR TITLE
Chromehtml2pdf

### PR DESCRIPTION
**Pull request by m-krajewski**
Support for Chromehtml2pdf engine.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+v1.0.0 NOT RELEASED
+============
+
+* introduced chromehtml2pdf renderer engine
+* removed support for wkhtmltopdf renderer engine
```
Comment (nowicki-maciej) [inline CHANGELOG.md:5]: Why did we remove support for wkhtmltopdf engine? I think both engines should be supported.

File: README.md [hunk 2]
```diff
@@ -4,9 +4,7 @@ The library for creating PDF files from html documents.
 
 ### System requirements ###
 
-This library requires `wkhtmltopdf 0.12.4 with patched Qt` to be installed on your machine. It should be added to system path. You can download it from [here](https://wkhtmltopdf.org/).
-
-`WARNING: Some Linux distributions may have old, not patched versions of wkhtmltopdf in their repositiories.`
+This library requires `chromehtml2pdf` to be installed on your machine. It should be added to system path. You can download it from [here](https://www.npmjs.com/package/chromehtml2pdf).
```
Comment (nowicki-maciej) [inline README.md]: 'to the system path'

File: README.md [hunk 3]
```diff
@@ -31,31 +29,52 @@ It is possible to customize the generated PDF using the following parameters:
...
-* No background
-* Grayscale
-* Enable/Disable smart shrinking
+* With background
+* Custom header and footer
 
 ### Setting parameters ###
 
-The following call will set portrait orientation, left margin, right margin and grayscale:
+The following call will set portrait page format, orientation, left margin, right margin and printing background:
```
Comment (nowicki-maciej) [inline README.md]: document background? probably sounds better

File: README.md [hunk 4]
```diff
@@ -31,31 +29,52 @@ It is possible to customize the generated PDF using the following parameters:
...
+Header and footer are loaded from files, which can contain just single valid html markup or proper html template with embedded styles.
+
+Following classes can be used to inject printing values into them:
+* date - formatted print date
+* title - document title
+* url - document location
+* pageNumber - current page number
+* totalPages - total pages in the document
+
+`Warning: Due to way chromehtml2pdf renders header and footer both of them needs to have specified font-size, otherwise it will not be visible.`
```
Comment (nowicki-maciej) [inline README.md]: they will not be visible

File: README.md [hunk 5]
```diff
@@ -31,31 +29,52 @@ It is possible to customize the generated PDF using the following parameters:
...
+            .withBackground()
         .and()
         .toPdf();
 ```
 
+### Custom header and footer options ###
+
+Header and footer are loaded from files, which can contain just single valid html markup or proper html template with embedded styles.
+
+Following classes can be used to inject printing values into them:
```
Comment (nowicki-maciej) [inline README.md]: html classes -> to avoid ambiguity

File: CHANGELOG.md (hunk 1 above)
Comment (tkaczmarzyk) [reply CHANGELOG.md:5]: there are some unresolved issues, which will be fixed later
please switch the release version to 0.1.0 though (1.0.0 will contain both chrome and wkthml2pdf)

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,9 @@
...
+
 v0.0.1 NOT RELEASED
```
Comment (tkaczmarzyk) [inline CHANGELOG.md:7]: add some basic description of this version

File: README.md [hunk 7]
```diff
@@ -31,31 +29,52 @@ It is possible to customize the generated PDF using the following parameters:
...
 ```java
 Document.fromStaticHtml("<html>")
         .and()
         .withParameters()
+            .pageFormat("A5")
             .portrait()
             .marginLeft("2cm")
             .marginRight("2cm")
-            .grayscale()
+            .withBackground()
```
Comment (tkaczmarzyk) [inline README.md:59]: this looks ambigous to me. What kind of background should I expect? Color? Image? If color, then which one? Can I customeize it anyhow? Can we introduce more meaningful name of this method?

File: src/main/java/com/tratif/pdfgen/document/builders/ParameterBuilder.java [hunk 8]
```diff
@@ -15,6 +15,12 @@
  */
 package com.tratif.pdfgen.document.builders;
```
Comment (tkaczmarzyk) [inline src/main/java/com/tratif/pdfgen/document/builders/ParameterBuilder.java:16]: hmm, we should keep 2 implementations of a common interface: one for wkhtml2pdf and one for chrome (the former might be not used until we fix other issues). Right now you are just replacing it, we shouldn't waste the previous implementation

**Review by tkaczmarzyk — CHANGES_REQUESTED** [review]

**Pull request merged**